### PR TITLE
persist: split persistcli into a new crate

### PR DESCRIPTION
**Pull request by danhhz**
In #20954, we get a circular dep from mz_persist_client (dev-deps) to mz_persist_txn to mz_persist_client. Apparently this is not a great idea, even though cargo allows it. Concretely, my r-a seems to be crashing noticeably less when I comment out the circular dep in the txn branch.

Context at https://github.com/rust-lang/rust-analyzer/issues/14167

We've got another one of these in mz_expr and mz_expr_test_util, but I'll leave that for someone else.

### Motivation

   * This PR refactors existing code.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/images/jobs/Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ MZFROM ubuntu-base
 
 RUN apt-get update && apt-get -qy install ca-certificates
 
-COPY examples/persistcli /usr/local/bin/persistcli
+COPY persistcli /usr/local/bin/persistcli
```
Comment (bkirwi) [inline misc/images/jobs/Dockerfile:14]: Is this covered by CI?

**Review by bkirwi — APPROVED** [review]
Code changes look good. Build changes look _plausible_, though I'm not sure if they're tested automatically... assuming you can either confirm that or list the manual testing this needs here, looks good to go!

File: misc/images/jobs/Dockerfile (hunk 1 above)
Comment (danhhz) [reply misc/images/jobs/Dockerfile:14]: Yup. That's how I caught it, in fact. Also verified via a `./bin/mzimage run --dev jobs -- --help`

**Review by danhhz — COMMENTED** [review]
TFTR!

> Build changes look plausible, though I'm not sure if they're tested automatically... assuming you can either confirm that or list the manual testing this needs here, looks good to go!

I feel pretty okay about this. Manual `bin/mzimage` testing confirms both `jobs` (which I forgot to initially fix and CI caught) and `maelstrom-persist` continue to work. CI uses `maelstrom-persist` in the various maelstrom tests and seems to be happy, so that's additional verification there. Also we got opted in to a nightlies run here, which has additional maelstrom tests as well as unused-deps

**Pull request merged**